indico/web/forms/fields/principals.py

```diff
@@ -53,6 +53,52 @@ def serialize_principal(principal):
 class PrincipalListField(HiddenField):
     """A field that lets you select a list Indico user/group ("principal")
 
+    :param allow_groups: If groups should be selectable.
+    :param allow_registration_forms: If registration form associated
+                                        to an event should be selectable.
+    :param allow_event_roles: If event roles should be selectable.
+    :param allow_category_roles: If category roles should be selectable.
+    :param allow_external_users: If "search users with no indico account"
+                                 should be available.  Selecting such a user
+                                 will automatically create a pending user once
+                                 the form is submitted, even if other fields
+                                 in the form fail to validate!
+    """
+
+    widget = JinjaWidget('forms/principal_list_widget.html', single_kwargs=True)
+
+    def __init__(self, *args, **kwargs):
+        self.data = []
```

Comment:
It doesn't seem to be initialized by the Field class, I added it previously to silence a warning. 
Still, this doesn't make much sense because `_get_data` below checks if this is null. 
But now I'm wondering, perhaps we could add it?

Reply:
isn't the warning just about it not being set in the constructur? ;)

Reply:
Yes, it is just a warning. But perhaps we could avoid the null check in the `get_data` method below if we directly initialised this as zero. Unless something else depends on this initial value?

Reply:
honestly, i would not try to refactor wtforms components too much as long as they work... hopefully we can eventually replace more and more with react+webargs/marshmallow